Add basic attribute validation when generating from a database table

Orteko opened the pull request:
These are generated based on the column data as reported from the database.

For instance, assuming an example database structure as follows:

```sql
CREATE TABLE `country` (
  `code` char(2) NOT NULL,
  `name` varchar(255) NOT NULL,
  `continent_name` varchar(255) DEFAULT NULL,
  `upstream_id` int(11) DEFAULT NULL,
  PRIMARY KEY (`code`)
)
```

With this commit in place we will now automatically generate model rules as follows:

```diff
     public static $rules = [
-        'name' => 'required'
+        'name' => 'required|string|max:255',
+        'continent_name' => 'nullable|string|max:255',
+        'upstream_id' => 'nullable|integer'
     ];
```

Note that this only matches the most common column types - any others will simply generate no extra rules as per the original behaviour.

ajayinfyom commented:
@Orteko Thank you for your effort.
but I see there also added rules for timestamps
`
     'created_at' => 'nullable',
     'updated_at' => 'nullable',
     'deleted_at' => 'nullable'
`   
need to be fixed/skip that rules.

Orteko commented:
> @Orteko Thank you for your effort.
> but I see there also added rules for timestamps
> `'created_at' => 'nullable', 'updated_at' => 'nullable', 'deleted_at' => 'nullable'`
> need to be fixed/skip that rules.

This was already set up to take into account excluded fields however it looks like the default excluded_fields array on the model was never actually referenced by the existing code.

I've now fixed that bug so it will fall back to the class defaults if no config override exists as well as adding deleted_at into the defaults.

Orteko commented:
Hmm, i've just realised one further edge case on this.

The class defaults are never actually used when the config/laravel_generator.php file is published as that will override it.

So at this point we have two options:

A: Sync up the class defaults with the config defaults by adding the various timestamp fields to infyom.laravel_generator.options.excluded_fields in config/laravel_generator.php

B: Remove the timestamps from the class-level excluded fields and refactor $dont_require_fields to also add the timestamps from infyom.laravel_generator.timestamps assuming infyom.laravel_generator.timestamps.enabled = true

I'm happy to implement either option?

mitulgolakiya commented:
@Orteko thanks for your efforts. PR Merged.

@fabiomlferreira it's available now.
